src/widgets/modals/CashTransactionModal.js

```diff
@@ -160,80 +161,79 @@ export const CashTransactionModal = ({ onConfirm }) => {
+          </View>
+          <View style={localStyles.iconContainerStyle}>
+            <ChevronDownIcon />
+          </View>
+        </TouchableOpacity>
+        <TouchableOpacity style={localStyles.containerStyle} onPress={onPressAmount}>
+          <View style={localStyles.textContainerStyle}>
+            <Text style={localStyles.textStyle}>{amountText}</Text>
+          </View>
+          <View style={localStyles.iconContainerStyle}>
+            <PencilIcon />
+          </View>
+        </TouchableOpacity>
+        <PressReason />
+        <TouchableOpacity style={localStyles.containerStyle} onPress={onPressDescription}>
+          <View style={localStyles.textContainerStyle}>
+            <Text style={localStyles.textStyle}>{descriptionText}</Text>
+          </View>
+          <View style={localStyles.iconContainerStyle}>
+            <PencilIcon />
+          </View>
+        </TouchableOpacity>
+        <BottomCurrencyEditor
+          isOpen={isAmountModalOpen}
+          buttonText="Confirm"
+          value={amountBuffer}
+          placeholder={placeholderTextAmount}
+          onChangeText={onChangeAmount}
+          onConfirm={onSubmitAmount}
+        />
+        <BottomModalContainer
+          isOpen={isReasonModalOpen}
+          modalStyle={localStyles.bottomModalContainerStyle}
+        >
+          <GenericChoiceList
+            data={reasons}
+            keyToDisplay={CASH_TRANSACTION_KEYS.REASON}
+            onPress={onSubmitReason}
+            highlightValue={reason?.title}
+          />
+        </BottomModalContainer>
+        <BottomTextEditor
+          isOpen={isDescriptionModalOpen}
+          buttonText="Confirm"
+          value={descriptionBuffer}
+          placeholder={placeholderTextDescription}
+          onChangeText={onChangeText}
+          onConfirm={onSubmitDescription}
+        />
+        <PageButton
+          text="OK"
+          onPress={onCreate}
+          isDisabled={!isValidTransaction}
+          disabledColor={WARM_GREY}
+          style={localStyles.okButton}
+          textStyle={localStyles.pageButtonTextStyle}
         />
-      </BottomModalContainer>
-      <BottomCurrencyEditor
-        isOpen={isAmountModalOpen}
-        buttonText="Confirm"
-        value={amountBuffer}
-        placeholder={placeholderTextAmount}
-        onChangeText={onChangeAmount}
-        onConfirm={onSubmitAmount}
-      />
-      <BottomModalContainer
-        isOpen={isReasonModalOpen}
-        modalStyle={localStyles.bottomModalContainerStyle}
-      >
-        <GenericChoiceList
-          data={reasons}
-          keyToDisplay={CASH_TRANSACTION_KEYS.REASON}
-          onPress={onSubmitReason}
-          highlightValue={reason?.title}
+      </View>
+      <ModalContainer title={placeholderTextName} isVisible={isNameModalOpen}>
+        <AutocompleteSelector
+          options={names}
+          queryString="name CONTAINS[c] $0"
+          sortKeyString="name"
```

Comment:
Would be good to use the `renderLeftText` and `renderRightText` I think. For a patient, using `${lastName}, ${firstName}` and for anyone else just `${name}`. Alternatively you could make the `Name.toString()` do the work for that.

A `renderRightText` for showing the `code` and possibly any other details - `D.O.B` maybe?

Reply:
Nice. 58567d32597e762d194d28447967defa54f4fe78, 87f4d1141f5eddf2d1625aa849a35fdeeaf68f80.